### DISCUSSION TITLE
Can we change view/edit template of a tiddler based on the tag?

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#6158] [Q&A] [answered]
We are creating an Anki-like feature in TW ([oflg/fishing](https://github.com/oflg/fishing) ), using the title field (or maybe a "question" field) as the question, and the text field as the answer.
And some control symbol `__xxx__` can make a text to be a cloze in the question. So I wonder if there is a way to make edit template show two codemirror editor when a tiddler have a specific tag added by our plugin, so user can edit title (or a question field) and answer field using codemirror and editor toolbar.

And maybe I also need to change the view template so user can have different QA card display by adding different tag to the tiddler.

Are we supposed to change view/edit template like this?

**Comment by linonetwo** [comment]
I think actually my need is

1. view template tiddler needs a "condition" field, so we can add UI element that only show up when we are viewing a QA flash card
2. we may need a way to allow edit any field with complete textarea (with toolbar and codemirror), maybe adding a editor template with a "condition" field is sufficient, so I can make flash card's question and answer field editable in the edit mode.

So I think I may need to add "condition" field support to both `tags: $:/tags/EditTemplate` and `tags: $:/tags/ViewTemplate`

**Reply by saqimtiaz** [reply]
@linonetwo do you want to show a different view template instead of the default, OR do you want to show some extra UI elements on QA tiddlers in addition to standard view template?

**Reply by linonetwo** [reply]
Just some extra UI element, for example, a hidden answer widget, onClick it will reveal the answer and some buttons to feedback the difficulty. This widget should only showup `in tiddler with "?" tag && tiddler title matches some state like $:/state/qa-current`. So I think adding a "condition" field will be good.

**Comment by saqimtiaz** [comment, marked as answer]
@linonetwo add your extra UI in a tiddler tagged $:/tags/ViewTemplate and wrap it in a $list widget with your condition:
```
<$list filter="[ ...mycondition ...]" variable="null">
<! custom UI here -->
</$list>
```
If you want to adjust where this UI is shown in comparison to other view template components, add `list-before` field to your template tiddler.

**Reply by linonetwo** [reply]
Thanks, I'm currently using reveal and list in different circumstance, this works like the `condition` field:

`<$reveal default={{$:/temp/fishing!!fish}} type="match" text=<<currentTiddler>> animate="yes">`

`<$list filter="[all[current]has[due]]" variable="null"> <div class="fishing-edit-title" /> </$list>`

And use `list-before: $:/core/ui/EditTemplate/title` to make it able to use sibling selector to hide other edit template, so I can replace them in some condition, this works likes `condition-not-show`:

```css
/** 
 * Hide title when editing a question.
 */
.fishing-edit-title + input.tc-edit-texteditor {
  display: none;
}
```

**Comment by saqimtiaz** [comment]
Adding more flexibility to how the templates are chosen is under discussion but is part of a larger considered change that is needed: https://github.com/Jermolene/TiddlyWiki5/issues/5136#issuecomment-735435268
